### PR TITLE
[CORE] Changes suggested by new llvm18 clang-format

### PR DESCRIPTION
**Pull request by smuzaffar**
This PR applies clang-format changes suggested by new llvm18 ( see https://github.com/cms-sw/cmssw/issues/45870 )

### CODE REVIEW

File: DataFormats/Common/interface/PtrVector.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ namespace edm {
 
     Ptr<T> const operator[](difference_type n) const {  // Otherwise the
       return base_->fromItr(iter_ + n);                 // boost::range
-    }                                                   // doesn't have []
+    }  // doesn't have []
```
Comment (makortel) [inline DataFormats/Common/interface/PtrVector.h:68]: In principle the new formatting is "worse" here, but really the entire multiline comment should be moved to a single line above the function definition. Better to be done after this PR.

**Comment by makortel** [comment]
+core

**Comment by makortel** [comment]
@smuzaffar Are you going to launch the tests for all of these formatting PRs, or should the reviewing L2s do that? (to avoid overloading the test nodes)

**Comment by smuzaffar** [comment]
@makortel , in order to not over load the build resources , I will wait for L2s to either start the tests or just sign these PRs (which should automatically start tests). I will start tests for remaining once jenkins has free resources

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**